.changeset/sour-squids-sip.md

```diff
@@ -0,0 +1,8 @@
+---
+'@clerk/clerk-sdk-node': patch
+'@clerk/backend': patch
+'@clerk/nextjs': patch
+'@clerk/remix': patch
+---
+
+Allow for a preferred source for the host attribute: default(host) | x-forwarded-host
```

Comment:
I think that `x-forwarded-host` should always have priority.
Before we move ahead with this PR, could you please share a repo where `clerkUrl` is correct while the headers are not?